src/lib/core/say.h

```diff
@@ -171,17 +171,12 @@ struct log {
 	pid_t pid;
 	/* Application identifier used to group syslog messages. */
 	char *syslog_ident;
-	/**
-	 * Used to wake up the main logger thread from a eio thread.
-	 */
-	ev_async log_async;
-	/**
-	 * Conditional variable securing variable below
-	 * from concurrent usage.
-	 */
-	struct fiber_cond rotate_cond;
 	/** Counter identifying number of threads executing log_rotate. */
 	int rotating_threads;
+	/** Mutex for critical sections. */
+	pthread_mutex_t mutex;
```

Comment:
Nit: Let's rename to `rotate_mutex` to match `rotate_cond` - after all it's used solely for log rotation. Also, please add a comment to `rotating_threads` that it's protected by this mutex.

Reply:
Done.